### PR TITLE
Refactor: compute metrics in experiments class

### PR DESCRIPTION
**Pull request by jxmorris12**
- Compute metrics in experiments class
  - Add PyTorch functions for metrics in `metrics.py` (that way metrics can run on GPU not CPU)
  - add metrics for Finetuning (todo: add metrics for Retrofitting)
  - remove logic for metric computations in `train.py`
- Support logging metrics multiple times per epoch via `--logs_per_epoch` arg
- Add loggers everywhere (instead of simple print statements); log to both file & stdout
- Add end-to-end tests for training
- Add `TensorRunningAverages` object that can average metrics that are logged multiple times, so we can log the average to w&b instead of logging them many times. (Also adds associated file `utils.py`)

### CODE REVIEW

**Comment by jscuds** [comment]
Closed because I merged `exp-metrics` into `rf-loss`
